tests/async/test_async_queryset.py

```diff
@@ -48,11 +57,11 @@ async def test_aiterator(self):
         self.assertCountEqual(results, [self.s1, self.s2, self.s3])
 
     async def test_aiterator_prefetch_related(self):
-        qs = SimpleModel.objects.prefetch_related("relatedmodels").aiterator()
-        msg = "Using QuerySet.aiterator() after prefetch_related() is not supported."
-        with self.assertRaisesMessage(NotSupportedError, msg):
-            async for m in qs:
-                pass
+        async for s in SimpleModel.objects.prefetch_related(
+            Prefetch("relatedmodel_set", to_attr="prefetched_relatedmodel")
```

Comment:
Using `to_attr` is a clever way of making sure the code effectively ran [without actually using `assertNumQueries`](https://github.com/django/django/pull/16543/files#r1115127795) 🎉